src/components/container/SeverityCalendar/SeverityCalendar.tsx

```diff
@@ -0,0 +1,158 @@
+import { useContext, useState } from "react";
+import { Guid, SurveyAnswer, SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+import React from "react";
+import queryAllSurveyAnswers from "../../../helpers/query-all-survey-answers";
+import Calendar from "../../presentational/Calendar/Calendar";
+import { format, parseISO, startOfDay, startOfMonth } from "date-fns";
+import { CalendarDay, CalendarDayStateConfiguration, DateRangeContext, LoadingIndicator } from "../../presentational";
+import { previewSeverityData } from "./SeverityCalendar.previewdata";
+import { useInitializeView } from "../../../helpers/Initialization";
+export type SeverityCalendarPreviewState = "Default" | "NoData";
+
+export interface SeverityCalendarProps {
+    surveyName: string,
+    dateRecordedResultIdentifier: string,
+    severityResultIdentifier: string,
+    intervalStart?: Date,
+    previewState?: SeverityCalendarPreviewState,
+}
+
+interface SeverityLogEntry {
+    dateObserved: Date,
+    severity: string,
+    dateEntered: string
+}
+
+export default function (props: SeverityCalendarProps) {
+    const surveyAnswerQuery: SurveyAnswersQuery = {
+        surveyName: props.surveyName,
+        resultIdentifier: [props.dateRecordedResultIdentifier, props.severityResultIdentifier]
+    }
+
+    const [data, setData] = useState<Map<string, SeverityLogEntry> | undefined>();
+    const dateRangeContext = useContext(DateRangeContext);
+    let intervalStart = dateRangeContext?.intervalStart ?? props.intervalStart ?? startOfMonth(new Date());
+
+    async function initialize() {
+        if (props.previewState) {
+            transformToCalendarData(props.previewState === "Default" ? previewSeverityData : []);
+        } else {
+            queryAllSurveyAnswers(surveyAnswerQuery).then((results: SurveyAnswer[]) => {
+                transformToCalendarData(results);
+            });
+        }
+    }
+
+    function transformToCalendarData(results: SurveyAnswer[]) {
+
+        var resultByDateMap = new Map<string, SeverityLogEntry>();
+        var groupedByResult = groupAnswersByResult(results);
+
+        groupedByResult.forEach((grouping) => {
+            var severity = "";
+            if (grouping.severityAnswer) {
+                switch (grouping.severityAnswer.answers[0]) {
+                    case "mild":
+                    case "severity1":
```

Comment:
Do we need to support these alternate severity values in this generic component?  They seem like they represent some project/survey specific implementation details.

I wonder if this calendar should support just `mild`, `moderate`, and `severe` by default, but then allow for an optional value mapping function prop in case a particular project wants to use values like `severity1`, `severity2`, and `severity3`.

Reply:
I created a severityValueMapper on the props